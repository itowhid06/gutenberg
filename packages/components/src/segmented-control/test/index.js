/**
 * External dependencies
 */
import { render } from '@testing-library/react';
import { act, Simulate } from 'react-dom/test-utils';

/**
 * Internal dependencies
 */
import { SegmentedControl } from '../index';

describe( 'SegmentedControl', () => {
	const options = [
		{ value: 'olaf', label: 'O' },
		{ value: 'samantha', label: 'S' },
	];

	it( 'should render correctly', () => {
		const { container } = render(
			<SegmentedControl
				baseId="segmented"
				options={ options }
				label="Test Segmented Control"
			/>
		);
		expect( container.firstChild ).toMatchSnapshot();
	} );

	it( 'should call onChange with proper value', () => {
		let container;
		const mockOnChange = jest.fn();
		act( () => {
			( { container } = render(
				<SegmentedControl
					baseId="segmented"
					options={ options }
					value={ options[ 1 ].value }
					onChange={ mockOnChange }
					label="Test Segmented Control"
				/>
			) );
		} );
		const firstButton = container.querySelector(
			`button[aria-label="${ options[ 0 ].label }"]`
		);
		act( () => {
			Simulate.click( firstButton );
		} );
		expect( mockOnChange ).toHaveBeenCalledWith( options[ 0 ].value );
	} );
} );
