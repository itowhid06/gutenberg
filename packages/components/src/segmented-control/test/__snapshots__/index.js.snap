// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`props should render correctly 1`] = `
.emotion-0 {
  background: #fff;
  border: 1px solid;
  border-color: #757575;
  border-radius: 2px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  min-height: 36px;
  min-width: 0;
  padding: 2px;
  position: relative;
  -webkit-transition: -webkit-transform 100ms linear;
  transition: transform 100ms linear;
}

@media ( prefers-reduced-motion: reduce ) {
  .emotion-0 {
    transition-duration: 0ms;
  }
}

.emotion-0:hover {
  border-color: #757575;
}

.emotion-0:focus {
  border-color: var( --wp-admin-theme-color-darker-10, #007cba);
  box-shadow: 0 0 0,0.5px,var( --wp-admin-theme-color, #00669b);
  outline: none;
  z-index: 1;
}

.emotion-3 {
  background: #1e1e1e;
  border-radius: 2px;
  box-shadow: transparent;
  left: 0;
  position: absolute;
  top: 2px;
  bottom: 2px;
  -webkit-transition: -webkit-transform 160ms ease;
  transition: transform 160ms ease;
  z-index: 1;
}

@media ( prefers-reduced-motion: reduce ) {
  .emotion-3 {
    transition-duration: 0ms;
  }
}

.emotion-5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  max-width: 100%;
  min-width: 0;
  position: relative;
}

.emotion-8 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: transparent;
  border: none;
  border-radius: 2px;
  color: #757575;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  line-height: 100%;
  outline: none;
  padding: 0 12px;
  position: relative;
  text-align: center;
  -webkit-transition: background 160ms linear,color 160ms linear,font-weight 60ms linear;
  transition: background 160ms linear,color 160ms linear,font-weight 60ms linear;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 100%;
  z-index: 2;
  color: #fff;
}

@media ( prefers-reduced-motion: reduce ) {
  .emotion-8 {
    transition-duration: 0ms;
  }
}

.emotion-8::-moz-focus-inner {
  border: 0;
}

.emotion-8:active {
  background: #fff;
}

.emotion-9 {
  font-size: 13px;
  line-height: 1;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate( -50%, -50% );
  -moz-transform: translate( -50%, -50% );
  -ms-transform: translate( -50%, -50% );
  transform: translate( -50%, -50% );
}

.emotion-11 {
  font-size: 13px;
  font-weight: bold;
  height: 0;
  overflow: hidden;
  visibility: hidden;
}

.emotion-13 {
  background: transparent;
}

.emotion-14 {
  background: rgba(0, 0, 0, 0.1);
  height: calc( 100% - 4px - 4px );
  position: absolute;
  right: 0;
  top: 4px;
  -webkit-transition: background 200ms linear;
  transition: background 200ms linear;
  width: 1px;
}

@media ( prefers-reduced-motion: reduce ) {
  .emotion-14 {
    transition-duration: 0ms;
  }
}

.emotion-19 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: transparent;
  border: none;
  border-radius: 2px;
  color: #757575;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  line-height: 100%;
  outline: none;
  padding: 0 12px;
  position: relative;
  text-align: center;
  -webkit-transition: background 160ms linear,color 160ms linear,font-weight 60ms linear;
  transition: background 160ms linear,color 160ms linear,font-weight 60ms linear;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 100%;
  z-index: 2;
}

@media ( prefers-reduced-motion: reduce ) {
  .emotion-19 {
    transition-duration: 0ms;
  }
}

.emotion-19::-moz-focus-inner {
  border: 0;
}

.emotion-19:active {
  background: #fff;
}

<div
  aria-label="Test Segmented Control"
  class="emotion-0 medium components-segmented-control emotion-1 emotion-2"
  data-wp-c16t="true"
  data-wp-component="SegmentedControl"
  id="segmented"
  role="radiogroup"
>
  <iframe
    aria-hidden="true"
    frameborder="0"
    src="about:blank"
    style="display: block; opacity: 0; position: absolute; top: 0px; left: 0px; height: 100%; width: 100%; overflow: hidden; pointer-events: none; z-index: -1;"
    tabindex="-1"
  />
  <div
    class="emotion-3 emotion-4"
    role="presentation"
    style="transform: translateX(NaNpx); transition: none; width: 0px;"
  />
  <div
    class="emotion-5 emotion-6 emotion-7"
    data-active="true"
  >
    <button
      aria-checked="true"
      aria-label="O"
      class="emotion-8"
      data-value="olaf"
      id="segmented-1"
      role="radio"
      tabindex="0"
    >
      <div
        class="emotion-9 emotion-10"
      >
        O
      </div>
      <div
        aria-hidden="true"
        class="emotion-11 emotion-12"
      >
        O
      </div>
    </button>
    <div
      aria-hidden="true"
      class="emotion-13 emotion-14 emotion-15"
    />
  </div>
  <div
    class="emotion-5 emotion-6 emotion-7"
    data-active="false"
  >
    <button
      aria-checked="false"
      aria-label="S"
      class="emotion-19"
      data-value="samantha"
      id="segmented-2"
      role="radio"
      tabindex="-1"
    >
      <div
        class="emotion-9 emotion-10"
      >
        S
      </div>
      <div
        aria-hidden="true"
        class="emotion-11 emotion-12"
      >
        S
      </div>
    </button>
    <div
      aria-hidden="true"
      class="emotion-13 emotion-14 emotion-15"
    />
  </div>
</div>
`;
